### PR TITLE
Local logistic regression

### PR DESCRIPTION
**Pull request by javinp**

### CODE REVIEW

File: src/main/java/org/bigml/binding/resources/AbstractResource.java [hunk 1]
```diff
@@ -64,64 +64,64 @@
...
-    static String TIMESERIES_RE = "^" + TIMESERIES_PATH + "/[a-f,0-9]{24}$";
-    static String FORECAST_RE = "^" + FORECAST_PATH + "/[a-f,0-9]{24}$";
-    static String DEEPNET_RE = "^" + DEEPNET_PATH + "/[a-f,0-9]{24}$";
-    static String OPTIML_RE = "^" + OPTIML_PATH + "/[a-f,0-9]{24}$";
-    static String FUSION_RE = "^" + FUSION_PATH + "/[a-f,0-9]{24}$";
+    public static String TIMESERIES_RE = "^" + TIMESERIES_PATH + "/[a-f,0-9]{24}$";
+    public static String FORECAST_RE = "^" + FORECAST_PATH + "/[a-f,0-9]{24}$";
+    public static String DEEPNET_RE = "^" + DEEPNET_PATH + "/[a-f,0-9]{24}$";
+    public static String OPTIML_RE = "^" + OPTIML_PATH + "/[a-f,0-9]{24}$";
+    public static String FUSION_RE = "^" + FUSION_PATH + "/[a-f,0-9]{24}$";
```
[the remark this answers is not in the document]
Comment (javinp) [reply src/main/java/org/bigml/binding/resources/AbstractResource.java]: Maybe those constants should be moved to Constants.java, as they can be used in other parts not only AbstractResource.

**Comment by sdesimone** [comment]
That could be an option — but, as a matter of fact, those members were not used outside of AbstractResource until now.

Actually, I see that most of those _RE members are not used anywhere.

Seems like cruft to me (speaking in terms of refactoring — meaning: I would not object to keeping things as they are (all members private; no one using them outside of AbstractResource - plenty of freedom to clean up things in the future). On the other hand, I would not like making that stuff public when it seems it is not adding real value (with the implications that once public, it would be much harder to get rid of them).

If you think we need an official way of checking if a received ID is a legitimate ID of a given kind, you could add that to AbstractResource.

For example:
boolean isLegitimateID(String ID) {
  //— trivial impl
}
(I would just check that the resource type matches [a-z]+ and the rest matches [a-f,0-9]{24})

**Comment by javinp** [comment]
It's not as easy, we have resources that can have several formats, for example

public static String CLUSTER_RE = "^(public/|)" + CLUSTER_PATH
            + "/[a-f,0-9]{24}$|^shared/" + CLUSTER_PATH
            + "/[a-zA-Z0-9]{26,27}$";

I'm going to revert those variables for now and I will redefine the value for LOGISTICREGRESSION_RE in LocalLogisticRegression. We can think later in refactoring, ok?

**Comment by sdesimone** [comment]
sure

**Pull request merged**